### PR TITLE
[changelog-parser] Add typing for changelog-parser

### PR DESCRIPTION
**Pull request by azerella**
@ungoldman Adding some type defs for `changelog-parser` ❤️ 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/changelog-parser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+     * Path to changelog file
+     */
+    filePath: string;
+    /**
+     * Changelog file string to parse
+     */
+    removeMarkdown: boolean;
+}
+
+declare function parseChangelog(options: Options|string, callback?: (result: string, error?: string) => void): Promise<object>;
```
Comment (43081j) [inline types/changelog-parser/index.d.ts]: this should probably be a `Partial<Options>` as you don't need to pass both options (you can pass `{ filePath: 'foo' }` for example).

Comment (adamzerella) [reply types/changelog-parser/index.d.ts]: @43081j Good pickup, I've updated the types based on your feedback 😄

File: types/changelog-parser/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     * Removes the markdown markup from the changelog entries by default.
+     * You can change its value to false to keep the markdown.
+     */
+    removeMarkdown: boolean;
+}
+
+/**
+ * Change log parser for node.
+ */
+declare function parseChangelog(options: Partial<Options>|string, callback?: (result: object, error: string|null) => void): Promise<object>;
```
Comment (43081j) [inline types/changelog-parser/index.d.ts]: last thing, is this the right way around? the callback. it seems to be:

`(err, result) => void` in tests
`(result, err) => void` here

need switching around maybe?

Comment (adamzerella) [reply types/changelog-parser/index.d.ts]: Updated test arguments, fixed

Comment (43081j) [reply types/changelog-parser/index.d.ts]: I'm super sorry to comment on this again but just to double check and be 100% certain, are we certain the error comes second?

https://github.com/hypermodules/changelog-parser/blob/master/README.md#callback

This seems to suggest a callback is indeed what the tests had originally: `(err: string|null, result: object) => void`

Comment (ungoldman) [reply types/changelog-parser/index.d.ts]: Hi @43081j, I am the author of `changelog-parser`, you are correct, the callback is error first.

Comment (43081j) [reply types/changelog-parser/index.d.ts]: yep though it looks like its merged already so i suppose a further PR will be needed to correct it. 

thanks for taking the time to confirm

**Pull request merged**